subsys/bluetooth/host/smp.c

```diff
@@ -277,6 +280,11 @@ static bool sc_supported;
 static const uint8_t *sc_public_key;
 static K_SEM_DEFINE(sc_local_pkey_ready, 0, 1);
 
+/* Pointer to internal data is used to mark that callbacks of given SMP channel are not initialized.
+ * Value of NULL represents no authenticaiton capabilities and cannot be used for that purpose.
+ */
+#define BT_SMP_AUTH_CB_UNINITIALIZED	((atomic_ptr_val_t)bt_smp_pool)
```

Comment:
I'm not sure I follow this. `BT_SMP_AUTH_CB_UNINITIALIZED` seems to be used to compare `smp->auth_cb`, but those are different structs. Why are we using this for comparison?

Reply:
I needed a memory address that would never be occupied by the application's authentication callbacks and decided to arbitrary use address of bt_smp_pool for that purpose.

Reply:
So you just needed a non-NULL, non-valid pointer to denote that the conn-specific auth CB was not initialized?

It seems like kind of a hack (even if documented properly). Wonder if it's worth to allocate another atomic variable for this specific reason instead.

Reply:
I would prefer to keep it as it is. I understand it could look hacky, but even if we allocate a dedicated variable, the types still would not match (and we would need some additional memory for the variable).
Keep in mind that pointing to SMP's internal data should be safe in this use-case.